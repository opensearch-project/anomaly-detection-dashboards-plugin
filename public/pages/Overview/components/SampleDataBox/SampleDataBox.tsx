/*
 * SPDX-License-Identifier: Apache-2.0
 *
 * The OpenSearch Contributors require contributions made to
 * this file be licensed under the Apache-2.0 license or a
 * compatible open source license.
 *
 * Modifications Copyright OpenSearch Contributors. See
 * GitHub history for details.
 */

import React from 'react';
import { get } from 'lodash';
import ContentPanel from '../../../../components/ContentPanel/ContentPanel';
import {
  EuiButton,
  EuiFlexGroup,
  EuiFlexItem,
  EuiTitle,
  EuiLink,
} from '@elastic/eui';
import { PLUGIN_NAME } from '../../../../utils/constants';

interface SampleDataBoxProps {
  title: string;
  icon: any;
  description: string;
  loadDataButtonDescription: string;
  onOpenFlyout(): void;
  onLoadData(): void;
  isLoadingData: boolean;
  isDataLoaded: boolean;
  detectorId: string;
  buttonDataTestSubj?: string;
}

export const SampleDataBox = (props: SampleDataBoxProps) => {
  return (
    <div style={{ height: 'auto' }}>
      <ContentPanel
        title={
          <div
            style={{
              display: 'flex',
              flexDirection: 'row',
              alignContent: 'flexStart',
            }}
          >
            {props.icon}
            <EuiTitle size="s">
              <h2 style={{ marginLeft: '12px', marginTop: '-3px' }}>
                {props.title}
              </h2>
            </EuiTitle>
            <EuiLink
              data-test-subj="flyoutInfoButton"
              style={{ marginLeft: '12px' }}
              onClick={props.onOpenFlyout}
            >
              Info
            </EuiLink>
          </div>
        }
        titleSize="s"
        badgeLabel={props.isDataLoaded ? 'INSTALLED' : undefined}
      >
        <EuiFlexGroup direction="column" gutterSize="l">
          <EuiFlexItem grow={false} style={{ height: '70px' }}>
            <p
              style={{
                textAlign: 'left',
                lineHeight: 1.4,
                maxHeight: 4.2,
              }}
            >
              {props.description}
            </p>
          </EuiFlexItem>
          <EuiFlexGroup
            style={{
              height: '100px',
              marginTop: '0px',
              marginBottom: '0px',
            }}
            direction="column"
            alignItems="center"
          >
            <EuiFlexItem grow={false}>
              <EuiButton
                style={{ width: '300px' }}
                data-test-subj={get(
                  props,
                  'buttonDataTestSubj',
                  'loadDataButton'
                )}
                disabled={props.isLoadingData || props.isDataLoaded}
                isLoading={props.isLoadingData}
                onClick={() => {
                  props.onLoadData();
                }}
              >
                {props.isLoadingData
                  ? 'Creating detector'
                  : props.isDataLoaded
                  ? 'Detector created'
                  : props.loadDataButtonDescription}
              </EuiButton>
            </EuiFlexItem>
            <EuiFlexItem grow={false}>
              {props.isDataLoaded ? (
                <EuiLink
                  data-test-subj="viewSampleDetectorLink"
                  href={`${PLUGIN_NAME}#/detectors/${props.detectorId}`}
                >
                  View detector and sample data
                </EuiLink>
              ) : null}
            </EuiFlexItem>
          </EuiFlexGroup>
        </EuiFlexGroup>
      </ContentPanel>
    </div>
  );
};
