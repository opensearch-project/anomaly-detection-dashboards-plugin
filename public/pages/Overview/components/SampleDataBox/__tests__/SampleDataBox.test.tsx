/*
 * SPDX-License-Identifier: Apache-2.0
 *
 * The OpenSearch Contributors require contributions made to
 * this file be licensed under the Apache-2.0 license or a
 * compatible open source license.
 *
 * Modifications Copyright OpenSearch Contributors. See
 * GitHub history for details.
 */

import React from 'react';
import { render } from '@testing-library/react';
import { EuiIcon } from '@elastic/eui';
import { SampleDataBox } from '../SampleDataBox';

const defaultProps = {
  title: 'Sample title',
  icon: <EuiIcon type="alert" />,
  description: 'Sample description',
  loadDataButtonDescription: 'Sample button description',
  onLoadData: jest.fn(),
  isLoadingData: false,
  isDataLoaded: false,
  detectorId: 'sample-detector-id',
};

describe('<SampleDataBox /> spec', () => {
  describe('Data not loaded', () => {
    test('renders component', () => {
      const { container, getByText } = render(
        <SampleDataBox {...defaultProps} />
      );
      expect(container.firstChild).toMatchSnapshot();
      getByText('Sample title');
      getByText('Sample description');
      getByText('Sample button description');
    });
  });
  describe('Data is loading', () => {
    test('renders component', () => {
      const { container, getByText } = render(
        <SampleDataBox {...defaultProps} isLoadingData={true} />
      );
      expect(container.firstChild).toMatchSnapshot();
      getByText('Sample title');
      getByText('Sample description');
      getByText('Creating detector');
    });
  });
  describe('Data is loaded', () => {
    test('renders component', () => {
      const { container, getByText } = render(
        <SampleDataBox {...defaultProps} isDataLoaded={true} />
      );
      expect(container.firstChild).toMatchSnapshot();
      getByText('Sample title');
      getByText('Sample description');
      getByText('Detector created');
    });
  });
});
