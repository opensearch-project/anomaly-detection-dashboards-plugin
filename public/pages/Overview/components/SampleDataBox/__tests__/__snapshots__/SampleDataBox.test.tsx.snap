// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SampleDataBox /> spec Data is loaded renders component 1`] = `
<div
  style="height: auto;"
>
  <div
    class="euiPanel euiPanel--paddingLarge euiPanel--borderRadiusMedium euiPanel--plain euiPanel--hasShadow euiCard euiCard--centerAligned euiCard--hasBetaBadge"
  >
    <div
      class="euiCard__content"
    >
      <span
        class="euiTitle euiTitle--small euiCard__title"
        id="random_html_idTitle"
      >
        <div
          style="display: flex; flex-direction: row; align-content: flexStart;"
        >
          <svg
            aria-hidden="true"
            class="euiIcon euiIcon--medium"
            focusable="false"
            height="16"
            role="img"
            viewBox="0 0 16 16"
            width="16"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M7.59 10.059 7.35 5.18h1.3L8.4 10.06h-.81Zm.394 1.901a.61.61 0 0 1-.448-.186.606.606 0 0 1-.186-.444c0-.174.062-.323.186-.446a.614.614 0 0 1 .448-.184c.169 0 .315.06.44.182.124.122.186.27.186.448a.6.6 0 0 1-.189.446.607.607 0 0 1-.437.184ZM2 14a1 1 0 0 1-.878-1.479l6-11a1 1 0 0 1 1.756 0l6 11A1 1 0 0 1 14 14H2Zm0-1h12L8 2 2 13Z"
              fill-rule="evenodd"
            />
          </svg>
          <h2
            class="euiTitle euiTitle--small"
            style="margin-left: 12px; margin-top: -3px;"
          >
            Sample title
          </h2>
          <button
            class="euiLink euiLink--primary"
            data-test-subj="flyoutInfoButton"
            style="margin-left: 12px;"
            type="button"
          >
            Info
          </button>
        </div>
      </span>
      <div
        class="euiCard__children"
      >
        <div
          class="euiFlexGroup euiFlexGroup--gutterLarge euiFlexGroup--directionColumn euiFlexGroup--responsive"
        >
          <hr
            class="euiHorizontalRule euiHorizontalRule--full euiHorizontalRule--marginSmall"
          />
          <div
            class="euiFlexItem euiFlexItem--flexGrowZero"
            style="height: 70px;"
          >
            <p
              style="text-align: left; line-height: 1.4; max-height: 4.2px;"
            >
              Sample description
            </p>
          </div>
          <div
            class="euiFlexGroup euiFlexGroup--gutterLarge euiFlexGroup--alignItemsCenter euiFlexGroup--directionColumn euiFlexGroup--responsive"
            style="height: 100px; margin-top: 0px; margin-bottom: 0px;"
          >
            <div
              class="euiFlexItem euiFlexItem--flexGrowZero"
            >
              <button
                class="euiButton euiButton--primary euiButton-isDisabled"
                data-test-subj="loadDataButton"
                disabled=""
                style="width: 300px;"
                type="button"
              >
                <span
                  class="euiButtonContent euiButton__content"
                >
                  <span
                    class="euiButton__text"
                  >
                    Detector created
                  </span>
                </span>
              </button>
            </div>
            <div
              class="euiFlexItem euiFlexItem--flexGrowZero"
            >
              <a
                class="euiLink euiLink--primary"
                data-test-subj="viewSampleDetectorLink"
                href="anomaly-detection-dashboards#/detectors/sample-detector-id"
                rel="noreferrer"
              >
                View detector and sample data
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
    <span
      class="euiCard__betaBadgeWrapper"
    >
      <span
        class="euiBetaBadge euiBetaBadge--hollow euiCard__betaBadge"
        id="random_html_idBetaBadge"
        title="INSTALLED"
      >
        INSTALLED
      </span>
    </span>
  </div>
</div>
`;

exports[`<SampleDataBox /> spec Data is loading renders component 1`] = `
<div
  style="height: auto;"
>
  <div
    class="euiPanel euiPanel--paddingLarge euiPanel--borderRadiusMedium euiPanel--plain euiPanel--hasShadow euiCard euiCard--centerAligned"
  >
    <div
      class="euiCard__content"
    >
      <span
        class="euiTitle euiTitle--small euiCard__title"
        id="random_html_idTitle"
      >
        <div
          style="display: flex; flex-direction: row; align-content: flexStart;"
        >
          <svg
            aria-hidden="true"
            class="euiIcon euiIcon--medium"
            focusable="false"
            height="16"
            role="img"
            viewBox="0 0 16 16"
            width="16"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M7.59 10.059 7.35 5.18h1.3L8.4 10.06h-.81Zm.394 1.901a.61.61 0 0 1-.448-.186.606.606 0 0 1-.186-.444c0-.174.062-.323.186-.446a.614.614 0 0 1 .448-.184c.169 0 .315.06.44.182.124.122.186.27.186.448a.6.6 0 0 1-.189.446.607.607 0 0 1-.437.184ZM2 14a1 1 0 0 1-.878-1.479l6-11a1 1 0 0 1 1.756 0l6 11A1 1 0 0 1 14 14H2Zm0-1h12L8 2 2 13Z"
              fill-rule="evenodd"
            />
          </svg>
          <h2
            class="euiTitle euiTitle--small"
            style="margin-left: 12px; margin-top: -3px;"
          >
            Sample title
          </h2>
          <button
            class="euiLink euiLink--primary"
            data-test-subj="flyoutInfoButton"
            style="margin-left: 12px;"
            type="button"
          >
            Info
          </button>
        </div>
      </span>
      <div
        class="euiCard__children"
      >
        <div
          class="euiFlexGroup euiFlexGroup--gutterLarge euiFlexGroup--directionColumn euiFlexGroup--responsive"
        >
          <hr
            class="euiHorizontalRule euiHorizontalRule--full euiHorizontalRule--marginSmall"
          />
          <div
            class="euiFlexItem euiFlexItem--flexGrowZero"
            style="height: 70px;"
          >
            <p
              style="text-align: left; line-height: 1.4; max-height: 4.2px;"
            >
              Sample description
            </p>
          </div>
          <div
            class="euiFlexGroup euiFlexGroup--gutterLarge euiFlexGroup--alignItemsCenter euiFlexGroup--directionColumn euiFlexGroup--responsive"
            style="height: 100px; margin-top: 0px; margin-bottom: 0px;"
          >
            <div
              class="euiFlexItem euiFlexItem--flexGrowZero"
            >
              <button
                class="euiButton euiButton--primary euiButton-isDisabled"
                data-test-subj="loadDataButton"
                disabled=""
                style="width: 300px;"
                type="button"
              >
                <span
                  class="euiButtonContent euiButton__content"
                >
                  <span
                    class="euiLoadingSpinner euiLoadingSpinner--medium euiButtonContent__spinner"
                  />
                  <span
                    class="euiButton__text"
                  >
                    Creating detector
                  </span>
                </span>
              </button>
            </div>
            <div
              class="euiFlexItem euiFlexItem--flexGrowZero"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<SampleDataBox /> spec Data not loaded renders component 1`] = `
<div
  style="height: auto;"
>
  <div
    class="euiPanel euiPanel--paddingLarge euiPanel--borderRadiusMedium euiPanel--plain euiPanel--hasShadow euiCard euiCard--centerAligned"
  >
    <div
      class="euiCard__content"
    >
      <span
        class="euiTitle euiTitle--small euiCard__title"
        id="random_html_idTitle"
      >
        <div
          style="display: flex; flex-direction: row; align-content: flexStart;"
        >
          <svg
            aria-hidden="true"
            class="euiIcon euiIcon--medium euiIcon-isLoading"
            focusable="false"
            height="16"
            role="img"
            viewBox="0 0 16 16"
            width="16"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M5.277 10.088c.02.014.04.03.057.047.582.55 1.134.812 1.666.812.586 0 1.84-.293 3.713-.88L9 6.212V2H7v4.212l-1.723 3.876Zm-.438.987L3.539 14h8.922l-1.32-2.969C9.096 11.677 7.733 12 7 12c-.74 0-1.463-.315-2.161-.925ZM6 2H5V1h6v1h-1v4l3.375 7.594A1 1 0 0 1 12.461 15H3.54a1 1 0 0 1-.914-1.406L6 6V2Z"
            />
          </svg>
          <h2
            class="euiTitle euiTitle--small"
            style="margin-left: 12px; margin-top: -3px;"
          >
            Sample title
          </h2>
          <button
            class="euiLink euiLink--primary"
            data-test-subj="flyoutInfoButton"
            style="margin-left: 12px;"
            type="button"
          >
            Info
          </button>
        </div>
      </span>
      <div
        class="euiCard__children"
      >
        <div
          class="euiFlexGroup euiFlexGroup--gutterLarge euiFlexGroup--directionColumn euiFlexGroup--responsive"
        >
          <hr
            class="euiHorizontalRule euiHorizontalRule--full euiHorizontalRule--marginSmall"
          />
          <div
            class="euiFlexItem euiFlexItem--flexGrowZero"
            style="height: 70px;"
          >
            <p
              style="text-align: left; line-height: 1.4; max-height: 4.2px;"
            >
              Sample description
            </p>
          </div>
          <div
            class="euiFlexGroup euiFlexGroup--gutterLarge euiFlexGroup--alignItemsCenter euiFlexGroup--directionColumn euiFlexGroup--responsive"
            style="height: 100px; margin-top: 0px; margin-bottom: 0px;"
          >
            <div
              class="euiFlexItem euiFlexItem--flexGrowZero"
            >
              <button
                class="euiButton euiButton--primary"
                data-test-subj="loadDataButton"
                style="width: 300px;"
                type="button"
              >
                <span
                  class="euiButtonContent euiButton__content"
                >
                  <span
                    class="euiButton__text"
                  >
                    Sample button description
                  </span>
                </span>
              </button>
            </div>
            <div
              class="euiFlexItem euiFlexItem--flexGrowZero"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
